WalletWasabi/WabiSabi/Crypto/CredentialRequesting/CredentialsResponseValidation.cs

```diff
@@ -7,7 +7,7 @@ namespace WalletWasabi.WabiSabi.Crypto.CredentialRequesting
 	/// Maintains the state needed to validate the credentials once the coordinator
 	/// issues them.
 	/// </summary>
-	public class CredentialsResponseValidation
+	public record CredentialsResponseValidation
```

Comment:
i think this entire file can just be `public record CredentialResponseValidation(Transcript Transcript, IEnumerable<Credential> Presented, IEnumerable<IssuanceValidationData> Requested)` but then where would the doc comments go?

Reply:
Yes, I think it could be done in that way but in case one day we extract and publish the WabiSabi library the internal constructor and transcript is better as it is.

Reply:
mostly for my C# learning: is that because the short syntax couples the protection of the constructor to that of the class and we want them at different levels? would it make sense to just an `internal record` instead

Reply:
It cannot be internal if we want to let others to import and use the library. But anyway, the think is that the shorter syntax is useful when to want to use the Deconstruct functionality (this is a c# 9 thing so we are still learning) because the order of the parameters is the order of the deconstructed tuple. I ean, it is not only for simplicity and more compact syntax, it is mostly for deconstructions.